### PR TITLE
🌿 fix(bridge): hide YOLO permission snapshots

### PR DESCRIPTION
**Pull request by AlexDochioiu**
## Complexity

🌿 Straightforward — this adds one localized configuration policy to the sole pending-permission snapshot handler and updates focused coverage.

## What

- Passes the bridge's YOLO setting into `GetSessionPermissionsHandler` as a required suppression policy.
- Returns an empty typed permission snapshot in YOLO mode after validating the request and before querying the backend.
- Adds regressions for snapshot suppression and preservation of empty-session validation.

## Why

YOLO already suppresses live permission events while auto-approving them, but a concurrent session-detail refresh could still read the backend permission before approval settled. Because the automatic reply event is also suppressed, the client could display a stale permission prompt that had already been approved.

## Risk and test focus

Low risk. The affected flow is limited to `POST /session/permissions` while YOLO is enabled. Tests verify normal mode still maps pending permissions, YOLO returns none even when the plugin reports one, and malformed requests still return 400. Auto-approval itself and session-family ordering remain unchanged.

## Expected result

- **User-visible:** Permission prompts no longer appear through session snapshots while YOLO mode is enabled.
- **Persisted/database:** No change.
- **Internal:** The live-event and snapshot exposure paths now enforce the same YOLO policy. No wire, client, analytics, or plugin-interface change.

## Verification

- TDD red: YOLO regression initially returned the plugin permission instead of an empty snapshot
- `dart test test/bridge/routing/get_session_permissions_handler_test.dart` — 10 passed
- Focused handler and auto-approval suites — 11 passed
- `dart analyze --fatal-infos` — no issues
- `dart test` — all 2,428 tests passed
- `git diff --check` — passed
- `aristotle-impl-review` — approved with no findings

---
## Summary by cubic
Hide pending permission snapshots when YOLO is enabled to prevent stale prompts during session refreshes and align with live-event suppression. Maintain backend error propagation while only suppressing the snapshot payload.

- **Bug Fixes**
  - Pass `config.yolo` into `GetSessionPermissionsHandler` as `suppressPendingPermissions`.
  - Query the repository first, then return an empty `PendingPermissionResponse` when suppression is on to preserve errors (e.g., not-found).
  - Keep validation: 400 on empty session ID; unknown sessions still raise plugin not-found.
  - Add tests for suppression, normal mapping, empty-session 400, and preserved not-found.

<sup>Written for commit be614780a0807c0675fa7fe5b6203a9dcabf25b3. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/sesori-ai/sesori_apps_monorepo/pull/738?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: bridge/app/lib/src/bridge/routing/get_session_permissions_handler.dart [hunk 1]
```diff
@@ -8,14 +8,18 @@ import "request_handler.dart";
...
-      );
+  GetSessionPermissionsHandler({
+    required PermissionRepository permissionRepository,
+    required bool suppressPendingPermissions,
+  }) : _permissionRepository = permissionRepository,
+       _suppressPendingPermissions = suppressPendingPermissions,
+       super(
+         HttpMethod.post,
+         "/session/permissions",
+         fromJson: SessionIdRequest.fromJson,
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply bridge/app/lib/src/bridge/routing/get_session_permissions_handler.dart:21]: [Sesori reply] Not addressed\n\nThe positional arguments are required by the pre-existing  constructor used across the routing hierarchy. Changing that base signature and every handler is unrelated broad cleanup for this localized bug fix; the new suppression policy itself is a required named parameter.

File: bridge/app/lib/src/bridge/routing/get_session_permissions_handler.dart [hunk 2]
```diff
@@ -30,6 +34,11 @@ class GetSessionPermissionsHandler extends BodyRequestHandler<SessionIdRequest,
       throw buildErrorResponse(request, 400, "empty session id");
     }
 
+    // YOLO suppresses permission prompts from both live events and this snapshot path.
+    if (_suppressPendingPermissions) {
+      return const PendingPermissionResponse(data: []);
+    }
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply bridge/app/lib/src/bridge/routing/get_session_permissions_handler.dart]: [Sesori reply] Addressed (in commit be614780)\n\nThe handler now performs the existing repository lookup before suppressing the payload, preserving unknown-session and plugin errors without adding a new repository API. Added a suppressed-mode unknown-session regression.

File: bridge/app/lib/src/bridge/routing/get_session_permissions_handler.dart [hunk 3]
```diff
@@ -30,6 +34,11 @@ class GetSessionPermissionsHandler extends BodyRequestHandler<SessionIdRequest,
       throw buildErrorResponse(request, 400, "empty session id");
     }
 
+    // YOLO suppresses permission prompts from both live events and this snapshot path.
+    if (_suppressPendingPermissions) {
```
[the remark this answers is not in the document]
Comment (AlexDochioiu) [reply bridge/app/lib/src/bridge/routing/get_session_permissions_handler.dart:40]: [Sesori reply] Addressed (in commit be614780)\n\nThe handler now performs the existing repository lookup before suppressing the payload, preserving unknown-session and plugin errors while still returning no permission data in YOLO mode. Added a suppressed-mode unknown-session regression.
